tests/Unit/NumericalAlgorithms/Spectral/Test_SwshTransformJob.cpp

```diff
@@ -0,0 +1,208 @@
+  static std::string name() noexcept { return "TestTag"; }
+  using type = Scalar<SpinWeighted<ComplexDataVector, Spin>>;
+};
+
+// for storing an expected spin-weighted value during the transform test
+template <int Spin>
+struct ExpectedTestTag : db::SimpleTag {
+  static std::string name() noexcept { return "ExpectedTestTag"; }
+  using type = Scalar<SpinWeighted<ComplexDataVector, Spin>>;
+};
+
+using TestDerivativeTagList =
+    tmpl::list<Tags::Derivative<TestTag<-1>, Tags::Eth>,
+               Tags::Derivative<TestTag<-1>, Tags::EthEthbar>,
+               Tags::Derivative<ExpectedTestTag<-1>, Tags::EthEthbar>,
+               Tags::Derivative<TestTag<2>, Tags::EthbarEthbar>>;
+
+/// [make_swsh_transform_job_list]
+using ExpectedInverseJobs = tmpl::list<
+    TransformJob<
+        -1, ComplexRepresentation::RealsThenImags,
+        tmpl::list<Tags::Derivative<TestTag<-1>, Tags::EthEthbar>,
+                   Tags::Derivative<ExpectedTestTag<-1>, Tags::EthEthbar>>>,
+    TransformJob<0, ComplexRepresentation::RealsThenImags,
+                 tmpl::list<Tags::Derivative<TestTag<-1>, Tags::Eth>,
+                            Tags::Derivative<TestTag<2>, Tags::EthbarEthbar>>>>;
+
+static_assert(
+    cpp17::is_same_v<
+        make_swsh_transform_job_list<ComplexRepresentation::RealsThenImags,
+                                     TestDerivativeTagList>,
+        ExpectedInverseJobs>,
+    "failed testing make_swsh_transform_job_list");
+/// [make_swsh_transform_job_list]
+
+/// [make_swsh_transform_from_derivative_tags]
+using ExpectedJobs =
+    tmpl::list<TransformJob<-1, ComplexRepresentation::Interleaved,
+                            tmpl::list<TestTag<-1>, ExpectedTestTag<-1>>>,
+               TransformJob<2, ComplexRepresentation::Interleaved,
+                            tmpl::list<TestTag<2>>>>;
+
+static_assert(
+    cpp17::is_same_v<
+        make_swsh_transform_job_list_from_derivative_tags<
+            ComplexRepresentation::Interleaved, TestDerivativeTagList>,
+        ExpectedJobs>,
+    "failed testing make_swsh_transform_job_list_from_derivative_tags");
+/// [make_swsh_transform_from_derivative_tags]
+
+static_assert(
+    cpp17::is_same_v<
+        typename tmpl::front<ExpectedInverseJobs>::CoefficientTagList,
+        tmpl::list<
+            Tags::SwshTransform<Tags::Derivative<TestTag<-1>, Tags::EthEthbar>>,
+            Tags::SwshTransform<
+                Tags::Derivative<ExpectedTestTag<-1>, Tags::EthEthbar>>>>,
+    "failed testing TransformJob");
+
+template <ComplexRepresentation Representation, int S>
+void test_transform_and_inverse_transform() noexcept {
+  // generate parameters for the points to transform
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> sdist{2, 7};
+  const size_t l_max = sdist(gen);
+  const size_t number_of_radial_points = 2;
+  UniformCustomDistribution<double> coefficient_distribution{-10.0, 10.0};
+
+  // create the variables for the transformations
```

Comment:
I like this comment!  But please use the same names in the comment as in the code below, so that the code is easier to follow.